crates/tuono/tests/cli_build.rs

```diff
@@ -216,3 +221,20 @@ fn build_fails_with_no_config() {
             "Cannot find tuono.config.ts - is this a tuono project?",
         ));
 }
+
+#[test]
+#[serial]
+fn is_uses_custom_main_when_present() {
+    let temp_tuono_project = TempTuonoProject::new();
+    temp_tuono_project.add_file_with_content("./src/main.rs", CUSTOM_MAIN_FILE);
+
+    let mut test_tuono_build = Command::cargo_bin("tuono").unwrap();
+    test_tuono_build
+        .arg("build")
+        .arg("--no-js-emit")
+        .assert()
+        .success();
+
+    let temp_tuono_main_rs_path = temp_tuono_project.path().join(".tuono/main.rs");
+    assert!(!temp_tuono_main_rs_path.exists());
```

Comment:
I'd just add an assertion that the `.tuono/router.rs` file exists to be sure that something got created